client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -66,30 +68,31 @@
         Preconditions.checkNotNull(serializer);
         this.stream = stream;
         this.controller = controller;
+        this.segmentSealedCallBack = this::handleLogSealed;
```

Comment:
Update the code based on the review comment.

`handleLogSealed `now grabs the lock inside of the method. This method can be triggered in the following two ways. It can be invoked by the next invocation of the write (or flush/close) or    triggered by the netty call back to the response processor with a segment sealed exception. The `handleLogSealed ` method triggered by either of the two approaches uses a lookup refreshSegmentStatusMap to keep track of the segments for which SegmentSealedException was handled i.e. update to new segments and their new keyranges. This prevents multiple execution of handleLogSealed for the same segment.